java6/src/main/java/io/papermc/paperclip/Main.java

```diff
@@ -30,21 +30,14 @@ public static void main(final String[] args) {
     }
 
     private static int getJavaVersion() {
-        final String version = System.getProperty("java.version");
+        final String version = System.getProperty("java.class.version");
         final String[] parts = version.split("\\.");
 
-        final String errorMsg = "Could not determine version of the current JVM";
         if (parts.length == 0) {
-            throw new IllegalStateException(errorMsg);
+            throw new IllegalStateException("Could not determine version of the current JVM");
         }
 
-        if (parts[0].equals("1")) {
-            if (parts.length < 2) {
-                throw new IllegalStateException(errorMsg);
-            }
-            return Integer.parseInt(parts[1]);
-        } else {
-            return Integer.parseInt(parts[0]);
-        }
+        // class file versions: 1.1 = 45, 1.2 = 46, ..., 18 = 62
+        return Integer.parseInt(parts[0]) - 44;
```

Comment:
It would be ideal if this returned the actual byte-code version instead. Rename the method aptly, and ensure all callers use the correct version numbers.

Using constants to correctly check for versions is also a good idea, e.g.:
```java
private static final int MAJOR_VERSION_18 = 62;

if (getMajorBytecodeVersion() < MAJOR_VERSION_18) {
    throw ...;
}
```